### PR TITLE
Lowercase supplied TargetGroup

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/corefx/issues/16362

### CODE REVIEW

File: eng/build.ps1 [hunk 1]
```diff
@@ -38,13 +38,14 @@ foreach ($argument in $PSBoundParameters.Keys)
...
-    "Release"    { $arguments += " /p:ConfigurationGroup=Release -configuration Release" }
-    "ExtraArgs"  { $arguments += " " + $ExtraArgs }
-    "Restore"    { $defaultargs = "-restore" }
-    "Test"       { $defaultargs = "-test -warnaserror:0" }
-    "InitTools"  { $defaultargs = "-restore -warnaserror:0" }
-    default      { $arguments += " /p:$argument=$($PSBoundParameters[$argument])" }
+    "Debug"       { $arguments += " /p:ConfigurationGroup=Debug -configuration Debug" }
+    "Release"     { $arguments += " /p:ConfigurationGroup=Release -configuration Release" }
+    "ExtraArgs"   { $arguments += " " + $ExtraArgs }
+    "TargetGroup" { $arguments += " /p:TargetGroup=$($PSBoundParameters[$argument].ToLower())"}
```
Comment (stephentoub) [inline eng/build.ps1]: May not matter, but do you want ToLowerInvariant instead?

Comment (ViktorHofer) [reply eng/build.ps1]: Good idea.

**Comment by ViktorHofer** [comment]
@dotnet-bot test OSX x64 Debug Build
@dotnet-bot test Windows x64 Debug Build

**Comment by TylerBrinkley** [comment]
I think this may have broke `build.cmd -- /p:CoreCLROverridePath=d:\git\coreclr\bin\Product\Windows_NT.x64.Release\` as specified in https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bits. Is the `--` no longer required?

**Comment by safern** [comment]
The `--` is no longer needed. Thanks for pointing out, I will update the docs.

**Pull request merged**